axum-login/src/service.rs

```diff
@@ -92,8 +106,46 @@ impl<S, Backend: AuthnBackend, Sessions: SessionStore> Layer<S>
         let login_manager = AuthManager {
             inner,
             backend: self.backend.clone(),
+            data_key: self.data_key,
         };
 
         self.session_manager_layer.layer(login_manager)
     }
 }
+
+/// Builder for the [`AuthManagerLayer`].
+#[derive(Debug, Clone)]
+pub struct AuthManagerLayerBuilder<Backend: AuthnBackend, Sessions: SessionStore> {
```

Comment:
> Change the AuthManagerLayer to be the builder itself. It would reduce the duplication of the fields between it and the new Builder.

This is basically what we're doing in `tower-sessions` now. I don't feel strongly one way or the other so I'm happy to go with whichever pattern you prefer.

Reply:
I'd leave it with this design, since it gives more wiggle room in the future. In my opinion this PR is mergable ;).

Reply:
Sounds good--I think the only thing stopping us from merging at the moment is a conflict.

Reply:
Done.